app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -747,6 +782,88 @@ public function importLead($fields, $data, $owner = null, $persist = true)
+            $dateIdentified = new DateTimeHelper($data[$fields['dateIdentified']]);
+            $lead->setDateIdentified($dateIdentified->getUtcDateTime());
+        }
+        unset($fields['dateIdentified']);
+
+        if (!empty($fields['createdByUser']) && !empty($data[$fields['createdByUser']])) {
+            $userRepo = $this->em->getRepository('MauticUserBundle:User');
+            $createdByUser = $userRepo->findByIdentifier($data[$fields['createdByUser']]);
+            if ($createdByUser !== null) {
+                $lead->setCreatedBy($createdByUser);
+            }
+        }
+        unset($fields['createdByUser']);
+
+        if (!empty($fields['modifiedByUser']) && !empty($data[$fields['modifiedByUser']])) {
+            $userRepo = $this->em->getRepository('MauticUserBundle:User');
+            $modifiedByUser = $userRepo->findByIdentifier($data[$fields['modifiedByUser']]);
+            if ($modifiedByUser !== null) {
+                $lead->setModifiedBy($modifiedByUser);
+            }
+        }
+        unset($fields['modifiedByUser']);
+
+        if (!empty($fields['ip']) && !empty($data[$fields['ip']])) {
+            $addresses = explode(',', $data[$fields['ip']]);
+            foreach ($addresses as $address) {
+                $ipAddress = new IpAddress;
+                $ipAddress->setIpAddress(trim($address));
+                $lead->addIpAddress($ipAddress);
+            }
+        }
+        unset($fields['ip']);
+
+        if (!empty($fields['points']) && !empty($data[$fields['points']]) && $lead->getId() === null) {
+            // Add points only for new leads
+            $lead->setPoints($data[$fields['points']]);
+
+            //add a lead point change log
+            $log = new PointsChangeLog();
+            $log->setDelta($data[$fields['points']]);
+            $log->setLead($lead);
+            $log->setType('lead');
+            $log->setEventName($this->factory->getTranslator()->trans('mautic.lead.import.event.name'));
+            $log->setActionName($this->factory->getTranslator()->trans('mautic.lead.import.action.name', array(
+                '%name%' => $this->factory->getUser()->getUsername()
+            )));
+            $log->setIpAddress($this->factory->getIpAddress());
+            $log->setDateAdded(new \DateTime());
+            $lead->addPointsChangeLog($log);
+        }
+        unset($fields['points']);
+
+        if (!empty($fields['doNotEmail']) && !empty($data[$fields['doNotEmail']]) && $hasEmail) {
+            $doNotEmail = filter_var($data[$fields['doNotEmail']], FILTER_VALIDATE_BOOLEAN);
+            if ($doNotEmail) {
+                $reason = $this->factory->getTranslator()->trans('mautic.lead.import.by.user', array(
+                    "%user%" => $this->factory->getUser()->getUsername()
+                ));
+                $this->setDoNotContact($lead, $data[$fields['email']], $reason);
```

Comment:
This causes `A new entity was found through the relationship 'Mautic\EmailBundle\Entity\DoNotEmail#lead' that was not configured to cascade persist operations for entity: Mautic\LeadBundle\Entity\Lead with ID #. To solve this issue: Either explicitly call EntityManager#persist() on this unknown entity or configure cascade persist  this association ...` error because the lead hasn't been saved to the DB yet. To get around this, we either have to define an association as we do the point logs (https://github.com/mautic/mautic/blob/staging/app/bundles/LeadBundle/Entity/Lead.php#L57 and associated functions) with at least a cascade={"persist"} or move the doNotEmail process to be after the lead is saved.  This will be an issue for scripts that use the importLead function with $persist = false unless we return both entities.  So I would be inclined to go the association route (be sure to use fetch="EXTRA_LAZY" as the point logs association does).